sequencer/src/bin/espresso-dev-node.rs

```diff
@@ -0,0 +1,332 @@
+
+    use async_compatibility_layer::{
+        art::async_sleep,
+        logging::{setup_backtrace, setup_logging},
+    };
+    use async_std::process::Command;
+    use committable::{Commitment, Committable};
+    use es_version::SequencerVersion;
+    use escargot::CargoBuild;
+    use portpicker::pick_unused_port;
+    use reqwest::StatusCode;
+    use sequencer::Transaction;
+    use sequencer_utils::AnvilOptions;
+
+    // If this test failed and you are doing changes on the following stuff, please
+    // sync your changes to [`espresso-sequencer-go`](https://github.com/EspressoSystems/espresso-sequencer-go)
+    // and open a PR.
+    // - APIs update
+    // - Types (like `Header`) update
+    #[async_std::test]
+    async fn dev_node_test() {
+        setup_logging();
+        setup_backtrace();
+        let anvil = AnvilOptions::default().spawn().await;
+        // let commitment_task_port = pick_unused_port().unwrap();
+        let commitment_task_port = 10002;
+        let api_port = pick_unused_port().unwrap();
+        let postgres_port = pick_unused_port().unwrap();
+
+        let mut db = Command::new("docker")
+            .arg("compose")
+            .arg("up")
+            .arg("-d")
+            .arg("sequencer-db-0")
+            .arg("--force-recreate")
+            .arg("--renew-anon-volumes")
+            .env("ESPRESSO_SEQUENCER0_DB_PORT", postgres_port.to_string())
+            .stdout(Stdio::null())
+            .spawn()
+            .unwrap();
+
+        let mut child_process = CargoBuild::new()
+            .bin("espresso-dev-node")
+            .features("testing")
+            .current_target()
+            .run()
+            .unwrap()
+            .command()
+            .env("ESPRESSO_SEQUENCER_L1_PROVIDER", anvil.url().to_string())
+            .env(
+                "ESPRESSO_COMMITMENT_TASK_PORT",
+                commitment_task_port.to_string(),
+            )
+            .env("ESPRESSO_SEQUENCER_API_PORT", api_port.to_string())
+            .env("ESPRESSO_SEQUENCER_POSTGRES_HOST", "localhost")
+            .env(
+                "ESPRESSO_SEQUENCER_POSTGRES_PORT",
+                postgres_port.to_string(),
+            )
+            .env("ESPRESSO_SEQUENCER_POSTGRES_USER", "root")
+            .env("ESPRESSO_SEQUENCER_POSTGRES_PASSWORD", "password")
+            .env("ESPRESSO_SEQUENCER_POSTGRES_DATABASE", "sequencer")
+            .spawn()
+            .unwrap();
+
+        let commitment_task_url = format!(
+            "http://localhost:{}/api/hotshot_contract",
+            commitment_task_port
+        );
+        println!("commitment task url: {}", commitment_task_url);
+
+        let sequencer_get_header_url =
+            format!("http://localhost:{}/availability/header/3", api_port);
+        println!("sequencer url: {}", sequencer_get_header_url);
+
+        // Waiting for the test node running completely
+        async_sleep(Duration::from_secs(100)).await;
```

Comment:
Can we replace this with something that waits for endpoints to be available instead of a sleep? The sleep might fail on slower hosts and waste time otherwise.

Reply:
Yeah I was thinking to call availability api

Reply:
Yeah or we could re-try to submit until that works.

Reply:
adjusted in b47398eb9a4970b61364f2f281681c6840a92131